src/Microsoft.DotNet.Watcher.Tools/Internal/MsBuildFileSetFactory.cs

```diff
@@ -162,29 +143,31 @@ public class MsBuildFileSetFactory : IFileSetFactory
             }
         }
 
-        // Ensures file exists in $(MSBuildProjectExtensionsPath)/$(MSBuildProjectFile).dotnetwatch.targets
-        private void EnsureInitialized()
+        private IReadOnlyList<string> InitializeArgs(string watchTargetsFile, bool trace)
         {
-            // see https://github.com/Microsoft/msbuild/blob/bf9b21cc7869b96ea2289ff31f6aaa5e1d525a26/src/XMakeTasks/Microsoft.Common.targets#L127
-            var projectExtensionFile = Path.Combine(_projectExtensionsPath,
-                Path.GetFileName(_projectFile) + ProjectExtensionFileExtension);
-
-            if (!File.Exists(projectExtensionFile))
+            var args = new List<string>
             {
-                // ensure obj folder is available
-                Directory.CreateDirectory(Path.GetDirectoryName(projectExtensionFile));
+                "/nologo",
+                "/v:n",
+                "/t:" + TargetName,
+                "/p:DotNetWatchBuild=true", // extensibility point for users
+                "/p:DesignTimeBuild=true", // don't do expensive things
+                "/p:CustomAfterMicrosoftCommonTargets=" + watchTargetsFile,
```

Comment:
You can't specify multiple targets to include in this property, can you? This prevents projects from using this extensibility point.

Reply:
Yes, this would override the project's setting. From what I've seen, this property is rarely used, much less than BaseIntermediateOutputPath. Even when used, it seems unlikely to change the list of files in Compile/EmbeddedResource/ProjectReference.

Another option is to write the targets file to [MSBuildUserExtensionsPath](https://github.com/Microsoft/msbuild/blob/1db4e2b95b2336d5b7083341ee0bff9cf0e3c533/src/Tasks/Microsoft.Common.CurrentVersion.targets#L28), though, I have reservations about adding files to machine state, and this wouldn't work across major version changes in MSBuild.

Reply:
Sounds good. I just wanted to make sure you considered the tradeoffs.

Reply:
Btw, before I merge this, are you aware of any other extensibility points I should consider? These are the ones I've looked at:
 - MSBuildProjectExtensionsPath 
 - MSBuildUserExtensionsPath
 - CustomAfterMicrosoftCommonTargets
 - CustomBeforeMicrosoftCommonTargets
 - $(MSBuildProjectFullPath).user

Reply:
`Directory.Build.targets` :trollface:

Reply:
haha. Going with CustomAfterMicrosoftCommonTargets for now.